tasks/collector.py

```diff
@@ -0,0 +1,97 @@
+import os
+import platform
+import subprocess
+import tempfile
+import urllib.request
+
+from invoke import task
+
+from tasks.go import tidy
+
+LICENSE_HEADER = """// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+"""
+
+
+@task(post=[tidy])
+def generate(ctx):
+    arch = platform.machine()
+    system = platform.system()
+    base_url = "https://github.com/open-telemetry/opentelemetry-collector/releases/download/cmd%2Fbuilder%2Fv0.103.1/"
+
+    if system == "Linux":
+        if arch == "x86_64":
+            binary_name = "ocb_0.103.1_linux_amd64"
+        elif arch == "arm64" or arch == "aarch64":
+            binary_name = "ocb_0.103.1_linux_arm64"
+        else:
+            print(f"Unsupported architecture: {arch}")
+            return
+    elif system == "Darwin":
+        if arch == "x86_64":
+            binary_name = "ocb_0.103.1_darwin_amd64"
+        elif arch == "arm64":
+            binary_name = "ocb_0.103.1_darwin_arm64"
+        else:
+            print(f"Unsupported architecture: {arch}")
+            return
+    else:
+        print(f"Unsupported system: {system}")
+        return
+
+    binary_url = f"{base_url}{binary_name}"
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        binary_path = os.path.join(tmpdirname, binary_name)
+        print(f"Downloading {binary_url} to {binary_path}...")
+
+        try:
+            urllib.request.urlretrieve(binary_url, binary_path)
+            os.chmod(binary_path, 0o755)
+            print(f"Downloaded to {binary_path}")
+        except Exception as e:
+            print(f"Failed to download {binary_url}: {e}")
+            return
+
+        # Run the binary with specified options
+        config_path = "./comp/otelcol/collector-contrib/impl/manifest.yaml"
+        run_command = f"{binary_path} --config {config_path} --skip-compilation"
+        print(f"Running command: {run_command}")
+
+        try:
+            result = subprocess.run(run_command, shell=True, check=True, capture_output=True, text=True)
+            print(f"Binary output:\n{result.stdout}")
+        except subprocess.CalledProcessError as e:
+            print(f"Failed to run the binary: {e}")
+            print(f"Error output:\n{e.stderr}")
+            return
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Same as above, an invoke Exit would be cleaner